src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -79,6 +89,11 @@ public class JsonMessageFormatter : IJsonRpcAsyncMessageTextFormatter
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         private Encoding encoding;
 
+        /// <summary>
+        /// Backing field for the <see cref="Rpc"/> property.
+        /// </summary>
+        private JsonRpc rpc;
+
         /// <summary>
```

Comment:
At this point, it appears the only use we have for this field is to access its `TraceSource`, which is a far looser coupling than `JsonRpc` itself. 
Can we swap this out for `TraceSource` (updating the container interface  and its name as well) so JsonRpc can pass it the TraceSource instead of itself?

Reply:
Well.... never mind. I guess the most important part of this field *isn't* the ability to access it as much as the ability to 'lock in' that it's exactly *one* instance that the formatter is associated with, which is crucial both for thread-safety and for the request IDs that we track.
Can you add a remarks section to this field then, explaining that this assists with these things?

Reply:
I see we're using it to send notifications, of course. :) Definitely need to keep it.